client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,461 @@
+  ) extends ClientState {
+    override def stateName: String = s"ConnectingExistingSession($sessionId)"
+    
+    override def handle(
+      event: StreamEvent,
+      transport: ZmqClientTransport,
+      config: ClientConfig,
+      serverRequestQueue: Queue[ServerRequest]
+    ): UIO[ClientState] = {
+      event match {
+        case StreamEvent.ServerMsg(SessionContinued(responseNonce)) =>
+          if (nonce == responseNonce) {
+            for {
+              _ <- ZIO.logInfo(s"Session continued: $sessionId")
+              now <- Clock.instant
+            } yield Connected(
+              sessionId = sessionId,
+              capabilities = capabilities,
+              createdAt = now,
+              serverRequestTracker = ServerRequestTracker(),
+              nextRequestId = nextRequestId,
+              pendingRequests = pendingRequests,
+              addresses = addresses
+            )
+          } else {
+            ZIO.logWarning("Nonce mismatch, ignoring SessionContinued").as(this)
+          }
+        
+        case StreamEvent.ServerMsg(SessionRejected(reason, responseNonce, leaderId)) =>
+          if (nonce == responseNonce) {
+            reason match {
+              case NotLeader =>
+                nextAddress match {
+                  case Some((address, newState)) =>
+                    for {
+                      _ <- ZIO.logInfo(s"Not leader, trying next: $address")
+                      newNonce <- Nonce.generate()
+                      _ <- transport.sendMessage(ContinueSession(sessionId, newNonce)).orDie
+                    } yield newState.copy(nonce = newNonce)
+                  case None =>
+                    for {
+                      _ <- ZIO.logInfo("No more addresses, retrying from start")
+                      newNonce <- Nonce.generate()
+                      _ <- transport.sendMessage(ContinueSession(sessionId, newNonce)).orDie
+                    } yield copy(currentAddressIndex = 0, nonce = newNonce)
+                }
+              
+              case SessionNotFound =>
+                ZIO.dieMessage("Session not found - cannot continue")
+              
+              case _ =>
+                ZIO.logWarning(s"Session rejected: $reason").as(Disconnected)
+            }
+          } else {
+            ZIO.logWarning("Nonce mismatch, ignoring SessionRejected").as(this)
+          }
+        
+        case StreamEvent.ServerMsg(other) =>
+          ZIO.logWarning(s"Unexpected message in ConnectingExistingSession: ${other.getClass.getSimpleName}").as(this)
+        
+        case StreamEvent.Action(ClientAction.SubmitCommand(_, promise)) =>
+          promise.fail(new IllegalStateException("Not connected")).ignore.as(this)
+        
+        case StreamEvent.Action(ClientAction.Disconnect) =>
+          ZIO.succeed(Disconnected)
+        
+        case _ =>
+          ZIO.succeed(this)
+      }
+    }
+    
+    def nextAddress: Option[(String, ConnectingExistingSession)] = {
```

Comment:
nextAddress should never turn None. It should use `nextIndex % address.Length` and roll to the next one.